### ISSUE TITLE
False negative with incompatible return type in subclass

### ISSUE THREAD

**Issue by raskama** [microsoft/pyright#10100]
Consider the following example:

```python
class A:
        @property
        def value(self) -> str:
                return "asdf"

class B(A):
        @property
        def value(self) -> int:
                return 1

def length_of_value(x: A):
        len(x.value)

length_of_value(B())
```

Pyright doesn't see any problem (running from cmd, v 1.1.396) while running the code produces well-deserved `TypeError: object of type 'int' has no len()`.

Giving the `value` property of B different return type than has the `value` property of A seems to be violating the Liskov substitution principle, but Pyright does not complain.

Mypy in this case produces the following error `error: Signature of "value" incompatible with supertype "A"`.

Here are some past issues which seem to target some more specific similar problems:
https://github.com/microsoft/pyright/issues/8167
https://github.com/microsoft/pyright/issues/8622

**Comment by erictraut** [comment]
Pyright correctly identifies the incompatible property override. You can see this in the [pyright playground](https://pyright-play.net/?code=MYGwhgzhAECCBcBYAUNN60AEAOAnA9tgKa4AuAnihhgCZEBm0AbmCAK5EAUERI9AlNAC0APmgRSuJKmqyMuIqTa4AdtABEkGvXUoUoSDABCnWP2lzoOAsTKUZsuoxbsuPPoNHQAlitIXLOQUlVWgARj1kJ2gQIhUAc1IACwB9fHoUlw5OAA94OHMqOViVXIA6LKJ%2BSJLE1PTM1myTfn4gA).

I'm guessing that you don't have the `reportIncompatibleMethodOverride` check enabled in your pyright configuration. This check is enabled by default in "standard" and "strict" type checking modes. See [this table](https://microsoft.github.io/pyright/#/configuration?id=diagnostic-settings-defaults) for details.

I'm going to close the issue because I don't think it's a bug. If you think my diagnosis is incorrect and can provide repro steps, let me know, and I'll dig into it further.
